egs/wsj/s5/utils/lang/limit_arpa_unk_history.py

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env python
+
+
+# Copyright 2018    Armin Oliya
+# Apache 2.0.
+
+
+# This script takes an existing ARPA lanugage model and limits the <unk> history 
+# to make it suitable for downstream <unk> modeling. This is for the case when
+# you don't have access to the original text corpus that is used for creating the LM.
+# If you do, you can use pocolm with the option --limit-unk-history=true. This
+# keeps the graph compact after adding the unk model.
+
+
+
+import argparse
+import codecs 
+import os
+import re 
+from collections import defaultdict
+
+
+
+parser = argparse.ArgumentParser(description='This script takes an existing ARPA lanugage model and limits the <unk> history to make it suitable for downstream <unk> modeling.')
+parser.add_argument('--old_lm',required=True, help='path to the old ARPA language model', type=str)
+parser.add_argument('--new_lm',required=True, help='path to the new ARPA model with limited unk history', type=str)
+
+args = parser.parse_args()	
+
+
+
+
+def get_ngram_stats(old_lm_lines):
+    ngram_counts = defaultdict(int) 
+
+    for i in range(10):
+        g = re.search(r"ngram (\d)=(\d+)",old_lm_lines[i])
+        if g:
+            ngram_counts[int(g.group(1))] = int(g.group(2))
+    
+    max_ngrams = list(ngram_counts.keys())[-1]
+    skip_rows = ngram_counts[1] 
+
+    return max_ngrams, skip_rows, ngram_counts
+
+
+def find_n_replace_unks(old_lm_lines,max_ngrams,skip_rows):
+
+    new_lm_lines = old_lm_lines[:skip_rows]
+    ngram_diffs =  defaultdict(int)
+
+    unk_pattern = re.compile("[0-9.-]+(?:[\s\\t]\S+){1,3}[\s\\t]<unk>[\s\\t](?!-[0-9]+\.[0-9]+).*")
```

Comment:
In general we don't assume that the written form of unk is `<unk>`: it may be capitalized , for instance. So it should be an option to the scripts.
And please make it _and_ instead of _n_.